### PR TITLE
docs: migrate wiki to docs site

### PR DESCRIPTION
**Pull request by tataihono**
# Description

Please include a summary of the change and which todo is completed. Please also include relevant motivation and context. List any dependencies that are required for this change.

- [migrate wiki to docs site](https://3.basecamp.com/3105655/buckets/29689605/todos/5521092765)

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] [Our Workflow](https://core-docs-git-22-08-tn-docs-migrate-wiki-jesusfilm.vercel.app/docs/engineering-practices/workflow/) should match our current process
- [ ] [I18n](https://core-docs-git-22-08-tn-docs-migrate-wiki-jesusfilm.vercel.app/docs/engineering-practices/internationalization) makes sense

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged into main

### CODE REVIEW

**Review by storyworks — DISMISSED** [review]

File: apps/docs/docs/04-engineering-practices/01-workflow/index.md [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+
+- [Reviewers: How to do a code review](https://google.github.io/eng-practices/review/reviewer/)
+- [Authors: getting through code review](https://google.github.io/eng-practices/review/developer/)
+
+As a reviewer doing a code review, you must make sure that:
+
+- The code is well-designed.
+- The functionality is good for the users of the code.
+- Any UI changes are sensible and look good.
+- Any parallel programming is done safely.
```
Comment (storyworks) [inline apps/docs/docs/04-engineering-practices/01-workflow/index.md:145]: What do you mean parallel programming?

File: apps/docs/docs/04-engineering-practices/01-workflow/index.md [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+- The code isn’t more complex than it needs to be.
+- The developer isn’t implementing things they might need in the future but doesn’t know they need now.
+- Code has appropriate unit tests.
+- Tests are well-designed.
+- The developer used clear names for everything.
+- Comments are clear and useful, and mostly explain why instead of what.
+- Code is appropriately documented (generally in g3doc).
```
Comment (storyworks) [inline apps/docs/docs/04-engineering-practices/01-workflow/index.md:152]: g3doc?

File: apps/docs/docs/04-engineering-practices/01-workflow/index.md [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+
+#### Address review comments
+
+Reviewers should leave questions, comments, and suggestions. Reviewers can comment on the whole pull request or add comments to specific lines. You and reviewers can insert images or code suggestions to clarify comments. For more information, see "[Reviewing changes in pull requests](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/reviewing-changes-in-pull-requests)."
+
+You can continue to commit and push changes in response to the reviews. Your pull request will update automatically.
+
+### Squash and merge your pull request
+
+Once your pull request is approved and all [checks have passed](#our-checks), squash and merge your pull request. This will automatically merge your branch so that your changes appear on the `main` branch. GitHub retains the history of comments and commits in the pull request to help future contributors understand your changes. For more information, see "[Merging a pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/merging-a-pull-request)."
```
Comment (storyworks) [inline apps/docs/docs/04-engineering-practices/01-workflow/index.md:163]: "squash and merge your pull request."

Squashing isn't manually done on our end is it?

File: apps/docs/docs/04-engineering-practices/01-workflow/index.md [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+
+This check runs `fetch-secrets`, `generate`, and `build` commands on all affected projects. This ensures that changes to our projects can still be built and deployed.
+
+### Tests
+
+This check runs `generate` and `test` on all affected projects. Typically the `test` command will run all integration, unit and e2e tests contained within the affected projects.
+
+### Visual Tests
+
+This check runs `build-storybook` on the shared-storybook project. It will then run the [Chromatic GitHub Action](https://www.chromatic.com/docs/github-actions) to automate our visual regression tests and publish [Storybook](https://storybook.core.jesusfilm.org/). Upon success two additional checks are added for [UI Tests](https://www.chromatic.com/docs/test) and [UI Review](https://www.chromatic.com/docs/review).
```
Comment (storyworks) [inline apps/docs/docs/04-engineering-practices/01-workflow/index.md:209]: Let's remove UI review and just do UI tests in future?

Comment (tataihono) [reply apps/docs/docs/04-engineering-practices/01-workflow/index.md:209]: we can remove this when the change actually gets made.

File: apps/docs/docs/04-engineering-practices/01-workflow/index.md (hunk 3 above)
Comment (tataihono) [reply apps/docs/docs/04-engineering-practices/01-workflow/index.md:163]: yes. You push the button.

File: apps/docs/docs/04-engineering-practices/01-workflow/index.md (hunk 2 above)
Comment (tataihono) [reply apps/docs/docs/04-engineering-practices/01-workflow/index.md:152]: removed.

File: apps/docs/docs/04-engineering-practices/01-workflow/index.md (hunk 1 above)
Comment (tataihono) [reply apps/docs/docs/04-engineering-practices/01-workflow/index.md:145]: removed.

**Review by storyworks — APPROVED** [review]

**Pull request merged**